### PR TITLE
[Feature] Search request reason field

### LINKED ISSUE #8368: Radio buttons: Reason for the talent search/request
## Gray's notes
Radio button on the search submit page will let the team better prioritise and save time.

Required radio selection

**Reason for the talent request:**
Select the option that best represents your reason for submitting this talent request
- Looking for immediate hire (approval, position and funding in place)
- For upcoming need (classification or funding not yet in place)
- General interest
- Required under the Directive on Digital Talent (pre-contracting talent search for Digital Services Contracting Questionnaire)

Learn more about the [Directive on Digital Talent](https://talent.canada.ca/en/directive-on-digital-talent).

## ✨ Feature

On the request page, add a field set where managers can share their staffing plans for the candidates they have requested from DCMO. They can do this using a separate form field designed to capture their answers to the following questions:

**Reason for the talent request:**
Select the option that best represents your reason for submitting this talent request
- Looking for immediate hire (approval, position and funding in place)
- For upcoming need (classification or funding not yet in place)
- General interest
- Required under the Directive on Digital Talent (pre-contracting talent search for Digital Services Contracting Questionnaire)

Learn more about the [Directive on Digital Talent](https://talent.canada.ca/en/directive-on-digital-talent).

On the admin (internal) facing side add new section populated by the selected radio

**Reason for the talent request**
(example) For upcoming need (classification or funding not yet in place) 

## 🕵️ Details
 - This has data capture implications for the Request responder interface. This information would need to be displayed in the information section of the request (management) page. 
 - Theoretically, each data set (for each question) could require its column and filter on the request table. 
 - A new model might need to be created to collect this data. 

[## 🎨 Design File External](https://www.figma.com/file/oUIFoHHAZ5FwFcGHc8p7Yt/Manager---Talent-request-form?type=design&node-id=330-14876&mode=design&t=8UBbbuxMvr9UZyni-4)

[## 🎨 Design File Internal](https://www.figma.com/file/dDo0sUzYFyplp7DvlLHKYJ/Admin---Requests-details?type=design&node-id=309-523&mode=design&t=x2ogOfOVGCemWFnm-4)

### 🧑‍🎨 Designer

@brookecameron

## 📸 Screenshot

external facing: 

Screenshot 2023-11-06 at 11 06 30 AM

Internal facing: 
Image

## ✅ Acceptance Criteria

A set of assumptions verifying that the feature was implemented correctly when tested.

- [ ] Manager can submit their staffing plans for their talent referral request.
- [ ] Recruitment  team can view, prioritize and report on staffing plans 
- [ ] New table field, add to model, schema, new enum
- [ ] Implement form on /submit
- [ ] Implement view on /admin

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #8368 

## 👋 Introduction

Adds a new field to the search request form for indicating the reason for submitting the request.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run migrations `docker-compose exec -w /home/site/wwwroot/api webserver sh -c "php artisan migrate:fresh --seed"`
2. Build `npm run dev`
3. Search for talent and continue to the request form
4. Confirm the new field exists and functions as expected
5. Submit the request
6. Login as admin
7. Navigate to the submitted request
8. Confirm the reason field appears

## 📸 Screenshot

Screenshot 2023-11-06 140256
Screenshot 2023-11-06 140315

### CODE REVIEW

File: api/app/Models/PoolCandidateSearchRequest.php [hunk 1]
```diff
@@ -29,6 +29,7 @@
  * @property Illuminate\Support\Carbon $updated_at
  * @property Illuminate\Support\Carbon $deleted_at
  * @property Illuminate\Support\Carbon $request_status_changed_at
+ * @property array $reason
```
Comment (vd1992) [inline api/app/Models/PoolCandidateSearchRequest.php:32]: This is just a comment so not controlling anything, but small typo, the field is string

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx [hunk 2]
```diff
@@ -259,6 +267,21 @@ export const ViewSearchRequest = ({
...
+          <FilterBlock
+            title={intl.formatMessage({
+              defaultMessage: "Reason for talent request",
+              id: "enffKD",
+              description: "Label for the reason for submitting the request.",
+            })}
+            content={
+              reason ? intl.formatMessage(getSearchRequestReason(reason)) : ""
+            }
+          />
```
Comment (vd1992) [inline apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx:279]: Less a specific ask and more a question if there is any rule of thumb for null state copy. We've added null state copy here and there but not without any specific rhythm as far as I can tell. 

So I was just wondering if this was fine for null state, so all requests submitted before this goes live. 
@BrookeCameron 

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Overall appears to work fantastic, just a small typo and a question below

File: api/app/Models/PoolCandidateSearchRequest.php (hunk 1 above)
Comment (esizer) [reply api/app/Models/PoolCandidateSearchRequest.php]: oops! I started with an array for some reason and forgot to change this! 😛 

Fixed in [fix typo in property comment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8408/commits/ae0a9c9a487e89f30780293bdb4e9f084a2a44ff)

File: apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx (hunk 2 above)
Comment (BrookeCameron) [reply apps/web/src/pages/SearchRequests/ViewSearchRequestPage/components/ViewSearchRequest.tsx:279]: as long as its **only** for requested submitted before this is live, then I don't see an issue with it.

**Review by vd1992 — APPROVED** [review]
💯

**Pull request merged**